src/sprout/Controllers/Admin/FileAdminController.php

```diff
@@ -480,29 +488,60 @@ public function ajaxDragdropSave()
             if ($size[0] > $max_dims['width'] or $size[1] > $max_dims['height']) {
                 $temp_path = STORAGE_PATH . 'temp/original_image_' . time() . '_' . Sprout::randStr(4);
                 $temp_path .= '.' . File::getExt($filename);
+
                 $img = new Image($src);
                 $img->resize($max_dims['width'], $max_dims['height']);
                 $img->save($temp_path);
 
-                $result = File::putExisting($filename, $temp_path);
-                unlink($temp_path);
-                unlink($src);
+                // Update the dimensions info
+                $dimensions = @getimagesize($temp_path);
+                $update_fields['imagesize'] = $dimensions ? json_encode($dimensions) : '';
+                $update_fields['filesize'] = filesize($temp_path);
+
+                try {
+                    Pdb::update('files', $update_fields, ['id' => $file_id]);
+                } catch (QueryException $e) {
+                    Json::error('Database error');
+                }
+
+                // Save the file to the backend
+                try {
+                    $result = File::putExisting($filename, $temp_path);
+                } catch (Exception $e) {
+                    Kohana::logException($e);
+                    Json::error($e);
+                }
+
+                File::cleanupLocalCopy($temp_path);
+                File::cleanupLocalCopy($src);
```

Comment:
This should be wrapped in a `finally {}` block.

Perhaps something like:

```php
try {
  Pdb::update(...);
  $result = File::putExisting(...);

} catch (QueryException $e) {
  Json::error('Database error');

} catch (Throwable $e) {
  Kohana::logException($e, false);
  Json::error($e);

} finally {
  File::cleanupLocalCopy($temp_file);
  File::cleanupLocalCopy($src);
}
```

Few extra things:
- catching Throwable so error types don't sneak though
- exception caught = false - because it's not, just wrapped in JSON